cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceQueryAdmin.java

```diff
@@ -103,11 +102,6 @@ public NamespaceMeta get(Id.Namespace namespaceId) throws Exception {
    */
   @Override
   public boolean exists(Id.Namespace namespaceId) throws Exception {
-    try {
-      get(namespaceId);
-    } catch (NotFoundException e) {
-      return false;
-    }
-    return true;
+    return nsStore.get(namespaceId) != null;
```

Comment:
Doesn't seem to be an easy way to fix this, other than silently ignoring ``UnauthorizedException`` from ``AbstractNamespaceClient.exists()``. Should ``exists`` even be exposed from an HTTP Client though?